core/os/thread.h

```diff
@@ -42,8 +42,6 @@
 #include "core/templates/safe_refcount.h"
 #include "core/typedefs.h"
 
-#include <new>
```

Comment:
FYI, this was actually needed and caused a crash with GCC 12: #100768 (fixed in #101004).
This was a honest mistake but shows we need to be somewhat cautious when reviewing IWYU suggestions.